pkg/rc/replication_controller.go

```diff
@@ -427,8 +474,14 @@ func (rc *replicationController) removePods(current types.PodLocations, eligible
 
 func (rc *replicationController) ensureConsistency(current types.PodLocations) error {
 	rc.mu.Lock()
+	// if a node transfer is in progress, the old node could be in current
+	// but could be unscheduled during this function's execution. We
+	// ignore it to prevent rescheduling
+	oldNode := rc.nodeTransfer.oldNode
```

Comment:
My previous change still allowed a race because a completed node transfer would make `rc.nodeTransfer.oldNode` nil. Then if the ineligible node was still in `current` it wouldn't be detected. I believe this fixes it